features/dashboard/lib/src/adding_report/ui/mobile/forest_adding_screen_mobile.dart

```diff
@@ -370,31 +391,24 @@ class _ForestAddingScreenMobileState extends State<ForestAddingScreenMobile> {
                                 crossAxisSpacing: 8,
                                 physics: const NeverScrollableScrollPhysics(),
                                 itemBuilder: (context, index) {
-                                  return Stack(
-                                      alignment: Alignment.topRight,
-                                      children: [
-                                        getImageWidget(_selectedImages,
-                                            widget.width * 0.4333)[index],
-                                        Padding(
-                                          padding: const EdgeInsets.only(
-                                              top: 5, right: 5),
-                                          child: GestureDetector(
-                                            onTap: () {
-                                              removeSelectedImage(index);
-                                            },
-                                            child: Icon(
-                                              Icons
-                                                  .remove_circle_outline_outlined,
-                                              color: Colors.white,
-                                              size: widget.width * 0.05,
-                                            ),
-                                          ),
-                                        )
-                                      ]);
+                                  return ImageGallery().buildPickerImage(
+                                      image: _selectedImages[index],
+                                      context: context,
+                                      width: widget.width * 2.4,
+                                      onRemoveTap: () {
+                                        removeSelectedImage(index);
+                                      });
                                 },
-                                itemCount: getImageWidget(
-                                        _selectedImages, widget.width * 0.4333)
-                                    .length,
+                                itemCount: _selectedImages.length,
+                              ),
+                            )
+                          : const SizedBox.shrink(),
+                      !isImagesSizeValid
+                          ? Text(
+                              'Nuotraukos per didelės',
```

Comment:
This message is not very clear for user. Consider message e.g. `Maksimalus nuotraukų dydis 10 MB`.